src/main/java/de/samply/common/ldmclient/centraxx/utils/PercentageLogger.java

```diff
@@ -0,0 +1,56 @@
+package de.samply.common.ldmclient.centraxx.utils;
+
+import org.slf4j.Logger;
+
+public class PercentageLogger {
+
+  private Logger logger;
+  private int numberOfElements;
+  private int counter = 0;
+  private int lastPercentage = 0;
+
+  /**
+   * Calculates percentage of remaining steps and logs it.
+   *
+   * @param logger logger
+   * @param numberOfElements number of elements
+   * @param description description of the operation
+   */
+  public PercentageLogger(Logger logger, int numberOfElements, String description) {
+
+    this.logger = logger;
+    this.numberOfElements = numberOfElements;
+    if (numberOfElements > 0) {
+      logger.debug(description);
+    }
+
+  }
+
+  /**
+   * Increments one step.
+   */
+  public void incrementCounter() {
+
+    if (numberOfElements > 0) {
+
+      counter++;
+
+      Double percentage = 100.0D * ((double) counter) / ((double) numberOfElements);
+      int ipercentage = percentage.intValue();
```

Comment:
just cast to int above, no need for double typed percentage var

Reply:
You are right. I tried to avoid losing information with the division (numberOfElements). However, 100 is at the beginning, and as far as we only want to show the percentage as an integer, we wouldn't lose information. If 100 comes at the end (... * 100) would be problematic, but not in this case.